chapters/ch20.md

```diff
@@ -1,49 +1,57 @@
 
 
 * Second, how much worse the algorithm does on the dev (or test) set than the training set. In this example, it does 1% worse on the dev set than the training set. We think of this informally as the algorithm’s **​variance** [6]
 ->
+Thứ hai, hiệu quả của thuật toán trên tập phát triển (hoặc kiểm tra) kém hơn bao nhiêu so với trên tập huấn luyện. Trong ví dụ này, thuật toán làm việc kém hơn 1% trên tập phát triển so với tập huấn luyện. Chúng ta tạm coi giá trị này như **phương sai** của thuật toán [6].
 
 
 Some changes to a learning algorithm can address the first component of error—​**bias​**—and improve its performance on the training set. Some changes address the second component—​**variance​**—and help it generalize better from the training set to the dev/test sets [7]. To select the most promising changes, it is incredibly useful to understand which of these two components of error is more pressing to address.
 ->
-
+Một số thay đổi trong thuật toán học có thể giải quyết thành phần thứ nhất của lỗi -- **độ chệch** -- và cải thiện hiệu quả của nó trên tập huấn luyện. Một số thay đổi giải quyết thành phần thứ hai -- **phương sai** -- và giúp thuật toán tổng quát hóa tốt hơn từ tập huấn luyện tới tập phát triển/kiểm tra [7]. Để lựa chọn thay đổi tiềm năng nhất, sẽ rất hữu ích khi hiểu thành nào trong hai thành phần lỗi là đáng để giải quyết hơn.
 
 Developing good intuition about Bias and Variance will help you choose effective changes for your algorithm.
 ->
-
+Phát triển một trực giác tốt về Độ chệch và Phương sai sẽ giúp bạn có những thay đổi hữu hiệu cho thuật toán.
 
 -----------------
 [6] The field of statistics has more formal definitions of bias and variance that we won’t worry about. Roughly, the bias is the error rate of your algorithm on your training set when you have a very large training set. The variance is how much worse you do on the test set compared to the training set in this setting. When your error metric is mean squared error, you can write down formulas specifying these two quantities, and prove that Total Error = Bias + Variance. But for our purposes of deciding how to make progress on an ML problem, the more informal definition of bias and variance given here will suffice.  
+->
+[6] Ngành thống kê có những định nghĩa chính thống hơn cho độ chệch và phương sai mà ở đây chúng ta không cần lưu tâm. Một cách <>, độ chệch là tỉ lệ lỗi thuật toán của bạn trên tập huấn luyện khi tập này rất lớn. Phương sai là sự giảm hiệu quả trên tập kiểm tra so với tập huấn luyện trong thiết lập này. Khi phép đo lỗi là trung bình bình phương lỗi, bạn có thể viết được công thứ tính hai đại lượng này, và chứng minh được rằng Tổng Lỗi = Độ Chệch + Phương Sai. Nhưng với mục đích xác định làm thế nào để tạo sự tiến triển trong một bài toán machine learning, định nghĩa ít chính thống hơn của độ chệch và phương sai như ở đây là đã đủ.
```

Comment:
1. "Một cách <>" ==> "Một cách xấp xỉ"
2. "công **thứ** tính hai đại lượng" ==> "công **thức** tính hai đại lượng"

Reply:
À, chỗ <> là mình chưa biết dùng từ nào. Từ này đồng nghĩa với "Đại loại" trong văn nói. Nhưng mình ít khi thấy "Đại loại" được dùng trong văn viết. Cụm "một cách xấp xỉ" không hợp lý trong trường hợp này, ít nhất tiếng Việt mình ít thấy người dùng như thế.

Để mình tìm thêm xem.

Reply:
Em có một vài candidate cho từ Roughly như: "đại thể, đại khái, phỏng chừng, áng chừng, ước chừng, đơn giản hoá" mà phần lớn đều nghe ko tự nhiên lắm. Theo hiểu thì cái này cũng giống như giới hạn trong toán, nếu số samples tiến tới vô cùng thì error tiến tới bias.

Reply:
Anh hiểu câu này là ý bác ấy đang định nghĩa lại nhưng một cách đại khái, không hẳn là liên quan tới giới hạn.